core/server/proxy/src/main/java/alluxio/proxy/s3/S3ErrorResponse.java

```diff
@@ -0,0 +1,170 @@
+    S3ErrorCode s3ErrorCode;
+    // TODO(WYY): we need to handle more exception in the future.
+    if (e instanceof alluxio.exception.status.NotFoundException) {
+      // 404
+      s3ErrorCode = S3ErrorCode.NO_SUCH_KEY;
+    } else if (e instanceof alluxio.exception.status.InvalidArgumentException) {
+      // 400
+      s3ErrorCode = S3ErrorCode.INVALID_ARGUMENT;
+    } else if (e instanceof alluxio.exception.status.PermissionDeniedException) {
+      // 403
+      s3ErrorCode = S3ErrorCode.ACCESS_DENIED_ERROR;
+    } else if (e instanceof alluxio.exception.status.FailedPreconditionException) {
+      // 412
+      s3ErrorCode = S3ErrorCode.PRECONDITION_FAILED;
+    } else {
+      // 500
+      s3ErrorCode = S3ErrorCode.INTERNAL_ERROR;
+    }
+    S3Error errorResponse = new S3Error(resource, s3ErrorCode);
+    errorResponse.setMessage(e.getMessage());
+    try {
+      return Response.status(s3ErrorCode.getStatus())
+          .entity(mapper.writeValueAsString(errorResponse)).build();
+    } catch (JsonProcessingException e2) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+          .entity("Failed to encode XML: " + e2.getMessage()).build();
+    } finally {
+      LOG.warn("mapper convert exception {} to {}.", e.getClass().getName(),
+          s3ErrorCode.getStatus().toString());
+    }
+  }
+
+  /**
+   * convert the IOException to the HTTP Response.
+   * @param e IOException
+   * @param resource resource
+   * @return response Http Response
+   */
+  private static Response createErrorResponse(IOException e, String resource) {
+    XmlMapper mapper = new XmlMapper();
+    S3ErrorCode s3ErrorCode;
+    // TODO(WYY): we need to handle more exception in the future.
+    if (e instanceof FileNotFoundException) {
+      // 404
+      s3ErrorCode = S3ErrorCode.NO_SUCH_KEY;
+    } else {
+      // 500
+      s3ErrorCode = S3ErrorCode.INTERNAL_ERROR;
+    }
+    S3Error errorResponse = new S3Error(resource, s3ErrorCode);
+    errorResponse.setMessage(e.getMessage());
+    try {
+      return Response.status(s3ErrorCode.getStatus())
+          .entity(mapper.writeValueAsString(errorResponse)).build();
+    } catch (JsonProcessingException e2) {
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+          .entity("Failed to encode XML: " + e2.getMessage()).build();
+    } finally {
+      LOG.warn("mapper convert exception {} to {}.", e.getClass().getName(),
+          s3ErrorCode.getStatus().toString());
+    }
+  }
+
+  /**
+   * convert the IOException to the HTTP Response.
+   * @param e AlluxioRuntimeException
+   * @param resource resource
+   * @return response Http Response
+   */
+  private static Response createErrorResponse(AlluxioRuntimeException e, String resource) {
+    XmlMapper mapper = new XmlMapper();
+    S3ErrorCode s3ErrorCode;
+    // TODO(WYY): we need to handle more exception in the future.
+    // 500
+    s3ErrorCode = S3ErrorCode.INTERNAL_ERROR;
```

Comment:
This is omitting the `NotFoundRuntimeException` which was added in https://github.com/Alluxio/alluxio/pull/15765
https://github.com/Alluxio/alluxio/blob/bf34b0e2efb83a1eef9677dc6f24830b31d84dec/core/server/proxy/src/main/java/alluxio/proxy/s3/S3ErrorResponse.java#L154-L161